### ISSUE TITLE
Bad positivity check in record module 

### ISSUE THREAD

**Issue by sattlerc** [agda/agda#4251]
On master (36b9448). Probably a duplicate of something.

```agda
postulate
  Path : {A : Set} → A → Set
  admit : {Z : Set} → Z
  X : Set
  Y : X → Set
  Y' : X → Set

record Square : Set₁ where
  field
    A : Set
    B : Set
    b : B
    coh : Path b

record Span : Set where
  record Cocone : Set where
    field
      x : X

  postulate
    m n : Cocone
    fung : (c : Cocone) → Y' (Cocone.x c) → Path c

  record CoconeDep (P : X → Set) : Set where
    inductive
    field
      x' : P (m .Cocone.x)

  s : Square
  s .Square.A = CoconeDep Y
  s .Square.B = Cocone
  s .Square.b = n
  s .Square.coh = fung _ admit
```
Error messsage:
```
CoconeDep is not strictly positive, because it occurs
in the first clause
in the definition of s, which occurs
                      in the first argument of Square.b
                      in the first argument of Cocone.x
                      (in the first argument of Y'
                       in the first argument of admit
                       in the third argument of fung
                       in the 4th clause
                       in the definition of s), which occurs
                                              in the first clause
                                              in the definition of Cocone.x, which occurs
                                                                           in an argument of a bound
                                                                           variable
                                                                           in the definition of
                                                                           CoconeDep.
```

**Comment by andreasabel** [comment]
In this concrete example, the record module mechanism is to be blamed, the problem goes away if you do not put the definition into the pseudo-record `Span`.  However, the same problem can be produced with a `where`-module.  The problem could be that with a where-module, the definitions are suddenly considered mutually recursive, changing the positivity analysis.  The problem is not reproducible by using the `mutual` keyword, as then the type checker complains first.  However, it is reproducible with a new-style mutual block.  
Looks like it is #1984.

Version with `where`-module:
```agda
test : Set₁
test = Set where

  record Cocone : Set where
    field
      x : X

  postulate
    m n : Cocone
    ...
```
Debug output:
```
Positivity graph: N=14 E=13
positivity graph for test Cocone Cocone.x m n fung CoconeDep
                     Issue4251.recCon-NOT-PRINTED CoconeDep.x' s
                     Issue4251.recCon-NOT-PRINTED
  Issue4251._.Cocone
    -[++]-> Issue4251._.s
      OccursWhere _ [] [InDefOf Issue4251._.s, InClause 1]
  Issue4251._.Cocone.x
    -[*]-> Issue4251._.CoconeDep
      OccursWhere _ [] [InDefOf Issue4251._.CoconeDep, VarArg]
    -[*]-> Issue4251._.fung.1
      OccursWhere _ [InDefOf Issue4251._.s,
                     InClause 3] [DefArg Issue4251._.fung 1, DefArg Issue4251.admit 0,
                                  DefArg Issue4251.Y' 0]
  Issue4251._.m
    -[++]-> Issue4251._.Cocone.x.0
      OccursWhere _ [InDefOf Issue4251._.CoconeDep,
                     VarArg] [DefArg Issue4251._.Cocone.x 0]
  Issue4251._.n
    -[++]-> Issue4251._.s
      OccursWhere _ [] [InDefOf Issue4251._.s, InClause 2]
  Issue4251._.fung
    -[++]-> Issue4251._.s
      OccursWhere _ [] [InDefOf Issue4251._.s, InClause 3]
  Issue4251._.CoconeDep
    -[++]-> Issue4251._.s
      OccursWhere _ [] [InDefOf Issue4251._.s, InClause 0]
  Issue4251._.CoconeDep.x'
  Issue4251._.s
    -[++]-> Issue4251._.Cocone.x.0
      OccursWhere _ [InDefOf Issue4251._.s, InClause 3,
                     DefArg Issue4251._.fung 1, DefArg Issue4251.admit 0,
                     DefArg Issue4251.Y' 0] [DefArg Issue4251._.Cocone.x 0,
                                             DefArg Issue4251.Square.b 0]
    -[++]-> Issue4251._.fung.0
      OccursWhere _ [InDefOf Issue4251._.s,
                     InClause 3] [DefArg Issue4251._.fung 0,
                                  DefArg Issue4251.Square.b 0]
  Issue4251._.Cocone.x.0
    -[++]-> Issue4251._.Cocone.x
      OccursWhere _ [] [InDefOf Issue4251._.Cocone.x, InClause 0]
  Issue4251._.fung.0
  Issue4251._.fung.1
  Issue4251._.CoconeDep.0
    -[++]-> Issue4251._.CoconeDep
      OccursWhere _ [] [InDefOf Issue4251._.CoconeDep]
  Issue4251._.CoconeDep.x'.0
    -[++]-> Issue4251._.CoconeDep.x'
      OccursWhere _ [] [InDefOf Issue4251._.CoconeDep.x', InClause 0]
  Issue4251._.s.0
    -[*]-> Issue4251._.s
      OccursWhere _ [] [InDefOf Issue4251._.s, InClause 0, Matched]
Positivity graph (completed): E=61
...
10 trivial sccs 1 non-trivial sccs with lengths [4]
  sccs = [[Issue4251._.Cocone.x, Issue4251._.CoconeDep, Issue4251._.s,
  Issue4251._.Cocone.x.0]]
setting Issue4251._.Cocone.x to (mutually) recursive
setting Issue4251._.CoconeDep to (mutually) recursive
setting Issue4251._.s to (mutually) recursive
setting Issue4251._.m to non-recursive
setting Issue4251._.n to non-recursive
setting Issue4251._.fung to non-recursive
setting Issue4251._.Cocone to non-recursive
Checking positivity of Cocone
record type  Cocone is not recursive
Checking positivity of CoconeDep
CoconeDep is not guarded, because it occurs
in the first clause
in the definition of s, which occurs...
```

**Comment by nad** [comment]
Issue #3592 might also be related.

**Comment by UlfNorell** [comment]
Closed this as duplicate of #1984
